keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -254,12 +263,138 @@ tried this feature in Alpha, we would have time to fix issues.
 - maxUnavailable greater than 1 with partition and staged pods greater than maxUnavailable
 - maxUnavailable greater than 1 with partition and maxUnavailable greater than replicas
 
+## Test Plan
+For `Alpha`, unit tests and e2e tests will be added to test functionality at both
+with feature flag enabled and disabled. Defaults will be verified so that users
+who donot set this flag are not surprised at all.
+
+
 ## Graduation Criteria
 
 - Alpha: Initial support for maxUnavailable in StatefulSets added. Disabled by default. 
 - Beta:  Enabled by default with default value of 1. 
 
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: MaxUnavailableStatefulSet
+  - Components depending on the feature gate: StatefulSet
+
+###### Does enabling the feature change any default behavior?
+
+No, the default behavior remains the same.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes this feature can be disabled. Once disabled, all existing  StatefulSet will
+revert to the old behavior where rolling update will proceed one pod at a time.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+If the feature is enabled and then later disabled, we will revert the value of 
+maxUnavailable to the default value so that customers are not surprised.
+
+###### Are there any tests for feature enablement/disablement?
+yes, there are unit tests which make sure the field is correctly dropped 
+on feature enable and disabled
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+MaxUnavailable can fail to rollout if the new image being deployed is bad and causes the 
+containers to crash. If there are not enough instances, it can bring down many or 
+cause disruption if the maxUnavailable is set to a large number, larger than what 
+the service can tolerate. 
+If the StatefulSet is already in a failed state with more than maxUnavailable pods 
+failing, changing maxUnavailable will not help, and it will require first fixing the 
+failing pod before the system can recover.
+
+It can impact already running workloads if service owner accidentally sets maxUnavailable
+to a very high number.
+
+###### What specific metrics should inform a rollback?
+The customer should look at the behavior of the maxUnavailable and if the rollout
+is not respecting the maxUnavailable, the feature should be rolledback.
```

Comment:
How can they look at it? metrics?

Reply:
added more specific data, they are not really metrics